File: app/src/main/java/com/zeoflow/emergency/service/EmergencyService.java
package com.zeoflow.emergency.service;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.media.AudioManager;
import android.media.MediaPlayer;
import android.media.SoundPool;
import android.net.Uri;
import android.os.Handler;
import android.os.IBinder;
import android.os.Looper;
import android.util.Log;

import com.zeoflow.emergency.R;

import static com.zeoflow.emergency.utils.Utils.STREAM_MUSIC;

public class EmergencyService extends Service
{

    @Override
    public IBinder onBind(Intent intent)
    {
        return null;
    }

    int sound;
    private SoundPool soundPool;
    float volume = 0.01f;
    private Runnable runnable;
    private int alarmMode = 1;

    @Override
    public int onStartCommand(Intent intent, int flags, int startId)
    {
        if (intent.getExtras() != null)
        {
            alarmMode = intent.getExtras().getInt("alarm");
        }
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onCreate()
    {

        soundPool = new SoundPool.Builder()
                .setMaxStreams(1)
                .build();

        int alarmSound = R.raw.alarm_1;
        if (alarmMode == 2)
        {
            alarmSound = R.raw.alarm_2;
        } else if (alarmMode == 3)
        {
            alarmSound = R.raw.alarm_3;
        }
        String path = "android.resource://" + getPackageName() + "/" + alarmSound;
        MediaPlayer mp = MediaPlayer.create(this, Uri.parse(path));
        long duration = mp.getDuration() / 100 + 1000;

        sound = soundPool.load(this, alarmSound, 1);

        soundPool.setOnLoadCompleteListener((soundPool, soundID, i1) ->
        {
            if (sound == soundID)
            {
                Handler mHandler = new Handler(Looper.getMainLooper());

                runnable = () ->
                {

                    AudioManager audioManager = (AudioManager) getApplicationContext().getSystemService(Context.AUDIO_SERVICE);
                    audioManager.setStreamVolume(STREAM_MUSIC, audioManager.getStreamMaxVolume(STREAM_MUSIC), 0);
                    soundPool.play(sound, volume, volume, 0, 0, 1F);
                    if (volume < 1f)
                    {
                        volume += 0.005f;
                    }
                    mHandler.postDelayed(runnable, duration);
                };
                mHandler.postDelayed(runnable, duration);
            }
        });
    }

    @Override
    public void onDestroy()
    {
        soundPool.release();
    }

}

File: app/src/main/java/com/zeoflow/emergency/service/EmergencyService.kt
package com.zeoflow.emergency.service

import android.app.Service
import android.content.Context
import android.content.Intent
import android.media.AudioManager
import android.media.MediaPlayer
import android.media.SoundPool
import android.net.Uri
import android.os.Handler
import android.os.IBinder
import android.os.Looper
import com.zeoflow.emergency.R
import com.zeoflow.emergency.utils.Utils.STREAM_MUSIC

class EmergencyService : Service() {

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    private var sound = 0
    private var soundPool: SoundPool? = null
    private var volume = 0.01f
    private var runnable: Runnable? = null
    private var alarmMode = 1
    private var duration: Long = 0
    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        if (intent.extras != null) {
            alarmMode = intent.extras!!.getInt("alarm")
        }
        var alarmSound = R.raw.alarm_1
        if (alarmMode == 2) {
            alarmSound = R.raw.alarm_2
        } else if (alarmMode == 3) {
            alarmSound = R.raw.alarm_3
        }
        val path = "android.resource://$packageName/$alarmSound"
        val mp = MediaPlayer.create(this, Uri.parse(path))
        duration = mp.duration / 100 * 2 + 1000.toLong()
        if (alarmMode == 2) {
            alarmSound = R.raw.alarm_2
        } else if (alarmMode == 3) {
            alarmSound = R.raw.alarm_3
        }
        sound = soundPool!!.load(this, alarmSound, 1)
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onCreate() {
        soundPool = SoundPool.Builder()
                .setMaxStreams(1)
                .build()

        soundPool?.setOnLoadCompleteListener { soundPool: SoundPool, soundID: Int, i1: Int ->
            if (sound == soundID) {
                val mHandler = Handler(Looper.getMainLooper())
                runnable = Runnable {
                    val audioManager = applicationContext.getSystemService(Context.AUDIO_SERVICE) as AudioManager
                    audioManager.setStreamVolume(STREAM_MUSIC, audioManager.getStreamMaxVolume(STREAM_MUSIC), 0)
                    soundPool.play(sound, volume, volume, 0, 0, 1f)
                    if (volume < 1f) {
                        volume += 0.005f
                    }
                    mHandler.postDelayed(runnable!!, duration)
                }
                mHandler.postDelayed(runnable!!, 0)
            }
        }
    }

    override fun onDestroy() {
        soundPool!!.release()
    }
}